### ISSUE TITLE
JavadocMethod validates methods of a private record with accessModifiers=public

### ISSUE THREAD

**Issue by arturict** [checkstyle/checkstyle#21697]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadocmethod.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac Rec2.java

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocMethod">
      <property name="accessModifiers" value="public"/>
    </module>
  </module>
</module>

$ cat Rec2.java
public class Rec2 {
    private record Hidden(int a) {
        /**
         * Doc.
         */
        public void method(int x) {
        }
    }

    private static class HiddenClass {
        /**
         * Doc.
         */
        public void method(int x) {
        }
    }
}

$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java $RUN_LOCALE -jar checkstyle-14.1.0-all.jar -c config.xml Rec2.java
Starting audit...
[ERROR] Rec2.java:6:32: Expected @param tag for 'x'. [JavadocMethod]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

Both methods sit in a private nested type, so with `accessModifiers=public` neither should be validated. The one in the private class is skipped as expected, the one in the private record is not.

`CheckUtil.getSurroundingAccessModifier` stops at `CLASS_DEF`, `INTERFACE_DEF`, `ANNOTATION_DEF` and `ENUM_DEF` but not at `RECORD_DEF`, so for a record it walks past it and returns the modifier of the public class around it. Found while working on #17117. #20338 changes the same method for compact sources.
